plugins/module_utils/network/sonic/config/lag_interfaces/lag_interfaces.py

```diff
@@ -445,6 +575,69 @@ def get_delete_lag_interfaces_requests(self, commands):
 
         return requests
 
+    def get_delete_po_ethernet_segment_requests(self, delete_portchannels, have):
+        es_commands = []
+        es_requests = []
+        es_path = 'data/openconfig-network-instance:network-instances/network-instance=default/evpn/ethernet-segments'
+        ess_path = es_path + '/ethernet-segment=%s'
+
+        for cmd in delete_portchannels:
+            po_name = cmd['name']
+            have_po = next((po for po in have if po['name'] == po_name), {})
+            have_es = have_po.get('ethernet_segment', {})
+            if have_es:
+                ed_ess_path = ess_path % quote(po_name, safe='')
+                es_request = {'path': ed_ess_path, 'method': DELETE}
+                es_requests.append(es_request)
+
+                es_commands.append(cmd)
+
+        return es_commands, es_requests
+
+    def get_delete_ethernet_segment_requests(self, delete_members, have, delete_es=False):
+        es_commands = []
+        es_requests = []
+        es_path = 'data/openconfig-network-instance:network-instances/network-instance=default/evpn/ethernet-segments'
+        ess_path = es_path + '/ethernet-segment=%s'
+        df_pref_path = es_path + '/ethernet-segment=%s/df-election/config/preference'
+
+        for cmd in delete_members:
+            po_name = cmd['name']
+            cmd_es = cmd.get('ethernet_segment', None)
+            if cmd_es:
+                have_po = next((po for po in have if po['name'] == po_name), {})
+                have_es = have_po.get('ethernet_segment', {})
+                if not have_es:
+                    continue
+
+                del_es = False
+                del_df_pref = False
+                if delete_es:
+                    del_es = True
+                else:
```

Comment:
This block should interpret a request to delete all three options as a request to delete the ethernet segment. It would also make sense for a deletion request that only specifies the esi_type to be interpreted as a request to delete the the ethernet segment.

Reply:
Yes!

Reply:
The changed code looks good.

Resolved.